rclpy/rclpy/executors.py

```diff
@@ -905,3 +905,21 @@ def spin_once_until_future_complete(
     ) -> None:
         future.add_done_callback(lambda x: self.wake())
         self._spin_once_impl(timeout_sec, future.done)
+
+    def shutdown(
+        self,
+        timeout_sec: float = None,
+        wait_for_threads: bool = True
```

Comment:
I'd recommend making this a [keyword only argument](https://peps.python.org/pep-3102/) to avoid `executor.shutdown(1.0, False)` looking ambiguous to a reader if the base class `shutdown` gets a second positional argument in the future.

```suggestion
        *,
        wait_for_threads: bool = True
```

Reply:
nice suggestion, thanks! fixed in https://github.com/ros2/rclpy/pull/1309/commits/43ce2d74e8636b523135163ad2dc8f9bf4b1f6f3